tests/yoti_common/third.party.attribute.converter.spec.js

```diff
@@ -0,0 +1,101 @@
+      const thirdPartyAttribute = ThirdPartyAttributeConverter
+        .convertThirdPartyAttribute(sampleThirdPartyAttribute);
+
+      expect(thirdPartyAttribute).toBeInstanceOf(AttributeIssuanceDetails);
+      expect(thirdPartyAttribute.getToken()).toEqual('someIssuanceToken');
+
+      expect(thirdPartyAttribute.getExpiryDate()
+        .getMicrosecondTimestamp()).toBe('2019-10-15T22:04:05.123000Z');
+
+      expect(thirdPartyAttribute.getIssuingAttributes().length).toEqual(1);
+      expect(thirdPartyAttribute.getIssuingAttributes()[0]).toEqual('com.thirdparty.id');
+    });
+
+    it('should return undefined for invalid third party attribute protobuf', () => {
+      const thirdPartyAttribute = ThirdPartyAttributeConverter.convertThirdPartyAttribute(Buffer.from('someValue', 'utf-8'));
+      expect(thirdPartyAttribute).toBe(undefined);
+    });
+
+    it('should return undefined for missing token', () => {
+      const thirdPartyProto = createTestThirdPartyAttribute(undefined, undefined);
+      const thirdPartyAttribute = ThirdPartyAttributeConverter
+        .convertThirdPartyAttribute(thirdPartyProto);
+      expect(thirdPartyAttribute).toBe(undefined);
+    });
+
+    it('should return undefined for empty token', () => {
+      const thirdPartyProto = createTestThirdPartyAttribute('', undefined);
+      const thirdPartyAttribute = ThirdPartyAttributeConverter
+        .convertThirdPartyAttribute(thirdPartyProto);
+      expect(thirdPartyAttribute).toBe(undefined);
+    });
+
+    it('should parse when issuing attributes is missing', () => {
+      const thirdPartyProto = createTestThirdPartyAttribute('someToken', undefined);
+      const thirdPartyAttribute = ThirdPartyAttributeConverter
+        .convertThirdPartyAttribute(thirdPartyProto);
+
+      expect(thirdPartyAttribute).not.toBe(undefined);
+      expect(thirdPartyAttribute).toBeInstanceOf(AttributeIssuanceDetails);
+      expect(thirdPartyAttribute.getToken()).toEqual('someToken');
+      expect(thirdPartyAttribute.getExpiryDate()).toBe(undefined);
+      expect(thirdPartyAttribute.getIssuingAttributes().length).toEqual(0);
+    });
+
+    it('should parse with invalid date', () => {
+      const thirdPartyProto = createTestThirdPartyAttribute('someToken', {
+        expiryDate: '2019-13-2',
+        definitions: [
+          {
+            name: 'com.thirdparty.id',
+          },
+        ],
+      });
+      const thirdPartyAttribute = ThirdPartyAttributeConverter
+        .convertThirdPartyAttribute(thirdPartyProto);
+
+      expect(thirdPartyAttribute).not.toBe(undefined);
+      expect(thirdPartyAttribute).toBeInstanceOf(AttributeIssuanceDetails);
+      expect(thirdPartyAttribute.getExpiryDate()).toBe(undefined);
+      expect(thirdPartyAttribute.getIssuingAttributes().length).toEqual(1);
+      expect(thirdPartyAttribute.getIssuingAttributes()[0]).toEqual('com.thirdparty.id');
+    });
+
+    it('should parse multiple attribute definitions', () => {
```

Comment:
Doesn't look like this test is finished - seems to be the same as the first lines of the test above?

Reply:
Fixed 👍